pkg/kubeapiserver/admission/configuration/configuration_manager.go

```diff
@@ -81,7 +94,30 @@ func (a *poller) notReady() {
 	a.ready = false
 }
 
+func (a *poller) bootstrapping() {
+	var ready bool
+	// bootstrapRetries is read-only, so no lock required.
+	for i := 0; i < a.bootstrapRetries; i++ {
+		a.lock.RLock()
+		ready = a.ready
+		a.lock.RUnlock()
+		if ready {
+			return
+		}
+		a.killTime()
+	}
+}
+
 func (a *poller) configuration() (runtime.Object, error) {
+	a.lock.Lock()
+	if !a.bootstrapped {
+		a.bootstrapped = true
```

Comment:
I really don't understand how this can work? shouldn't all callers be blocked until bootstrapping is finished?